### PR TITLE
fix: daylight saving issue

### PR DESCRIPTION
**Pull request by Mech0z**
## Description
The method being used was only for inserting data into the database, not for retrieving it.

## References

* #1851

### CODE REVIEW

**Review by x-platformcoder — APPROVED** [review]

File: source/iso8601/GreenEnergyHub.Iso8601/IIso8601Durations.cs [hunk 1]
```diff
@@ -30,6 +30,15 @@ public interface IIso8601Durations
         /// <returns>End time for time interval</returns>
         Instant AddDuration(Instant startInstant, string duration, int numberOfDurations);
 
+        /// <summary>
+        /// When displaying in search results in GUI an end-date needs to be calculated.
+        /// This date should not go beyond the 1. in the mont.
```
Comment (prtandrup) [inline source/iso8601/GreenEnergyHub.Iso8601/IIso8601Durations.cs:35]:
```suggestion
        /// When displaying price series results in frontend a valid to date needs to be calculated.
        /// This method has special handling for irregular monthly price series.
```

File: source/iso8601/GreenEnergyHub.Iso8601/IIso8601Durations.cs [hunk 2]
```diff
@@ -30,6 +30,15 @@ public interface IIso8601Durations
...
+        /// </summary>
+        /// <param name="startInstant"></param>
+        /// <param name="duration"></param>
+        /// <returns>End time for time interval</returns>
+        Instant AddDurationIrregularSupport(Instant startInstant, string duration);
```
Comment (prtandrup) [inline source/iso8601/GreenEnergyHub.Iso8601/IIso8601Durations.cs:40]:
```suggestion
        Instant AddDurationWithIrregularSupport(Instant startInstant, string duration);
```

Comment (Mech0z) [reply source/iso8601/GreenEnergyHub.Iso8601/IIso8601Durations.cs]: renamed

File: source/iso8601/GreenEnergyHub.Iso8601/IIso8601Durations.cs [hunk 3]
```diff
@@ -30,6 +30,15 @@ public interface IIso8601Durations
         /// <returns>End time for time interval</returns>
         Instant AddDuration(Instant startInstant, string duration, int numberOfDurations);
 
+        /// <summary>
+        /// When displaying price series results in frontend a valid to date needs to be calculated.
+        /// This method has special handling for irregular monthly price series.
+        /// </summary>
+        /// <param name="startInstant"></param>
+        /// <param name="duration"></param>
+        /// <returns>End time for time interval</returns>
```
Comment (prtandrup) [inline source/iso8601/GreenEnergyHub.Iso8601/IIso8601Durations.cs:39]:
```suggestion
        /// <returns>The calculated Valid To</returns>
```

**Review by prtandrup — APPROVED** [review]
approved with a minor comment thing

**Pull request merged**